iquip/apps/builder.py

```diff
@@ -0,0 +1,419 @@
+            name: See the attributes section in _BaseEntry.
+        """
+        super().__init__(parent=parent)
+        self.name = name
+        # widgets
+        self.nameLabel = QLabel(name, self)
+        # layout
+        self.layout = QHBoxLayout(self)
+        self.layout.addWidget(self.nameLabel)
+        self.setLayout(self.layout)
+
+    def value(self) -> Any:
+        """Returns the entered or selected value.
+        
+        This must be overridden in the subclass.
+        """
+        raise NotImplementedError
+
+
+class _BooleanEntry(_BaseEntry):
+    """Entry class for a boolean value."""
+
+    def __init__(self, name: str, default: bool = False, parent: Optional[QWidget] = None):
+        """Extended.
+        
+        Args:
+            default: The default value. If it does not exist, it is set to False.
+        """
+        super().__init__(name, parent=parent)
+        # widgets
+        self.checkBox = QCheckBox(self)
+        self.checkBox.setChecked(default)
+        # layout
+        self.layout.addWidget(self.checkBox)
+
+    def value(self) -> bool:
+        """Overridden.
+        
+        Returns the status of the checkBox.
+        """
+        return self.checkBox.isChecked()
+
+
+class _EnumerationEntry(_BaseEntry):
+    """Entry class for an enumeration value."""
+
+    def __init__(
+        self,
+        name: str,
+        choices: List[str],
+        default: Optional[str] = None,
+        parent: Optional[QWidget] = None
+    ):
+        """Extended.
+        
+        Args:
+            choices: The pre-defined candidates.
+            default: The default value. If it does not exist, it is set to the first candidate.
+        """
+        super().__init__(name, parent=parent)
+        # widgets
+        self.comboBox = QComboBox(self)
+        for choice in choices:
+            self.comboBox.addItem(choice)
+        if choices:
+            self.comboBox.setCurrentText(choices[0] if default is None else default)
+        # layout
+        self.layout.addWidget(self.comboBox)
+
+    def value(self) -> str:
+        """Overridden.
+        
+        Returns the value of the comboBox.
+        """
+        return self.comboBox.currentText()
+
+
+class _NumberEntry(_BaseEntry):
+    """Entry class for a number value.
+    
+    Attributes:
+        scale: The scale factor that actually applies.
+        type: The type of the value. If "int", value() returns an integer value.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        unit: str,
+        scale: float,
+        step: float,
+        min: Optional[float],  # pylint: disable=redefined-builtin
+        max: Optional[float],  # pylint: disable=redefined-builtin
+        ndecimals: int,
+        type: str,  # pylint: disable=redefined-builtin
+        default: Optional[float] = None,
+        parent: Optional[QWidget] = None
+    ):  # pylint: disable=too-many-arguments
+        """Extended.
+        
+        Args:
```

Comment:
I think you should clearly denote the unit or scale for each parameter (or write an explanation in this docstring). Remember that you were confused as well, and explain in more detail to prevent such confusion.

Reply:
It looks necessary! However, how about postponing it in the later issue?
I will implement a tooltip to show the real applied value and handle the pre-defined units.
I think the exact manual will be decided only after this is over.

Reply:
Okay then, please leave a TODO comment.